bitcoin/src/crypto/ecdsa/legacy.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Legacy ECDSA Bitcoin signatures.
+
+use core::fmt;
+use core::str::FromStr;
+
+use secp256k1;
+
+#[cfg(doc)]
+use crate::crypto::ecdsa::EcdsaSighashType;
+use crate::crypto::ecdsa::{self, Error, SerializedSignature};
+use crate::prelude::*;
+
+/// A legacy Bitcoin ECDSA signature.
+///
+/// This is just a thin wrapper around [`ecdsa::Signature`].
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Signature(pub(crate) ecdsa::Signature);
```

Comment:
I think you've misunderstood the point of the suggested changes. It definitely should carry the sighash flags so they can be put into witness stack correctly. You obtain the signature by signing a sighash which also carries the flags.